tsdb/store.go

```diff
@@ -89,6 +89,19 @@ func (s *Store) DeleteShard(shardID uint64) error {
 	return nil
 }
 
+// DeleteDatabase will close all shards associated with a database and remove the directory and files from disk.
+func (s *Store) DeleteDatabase(name string, shardIDs []uint64) error {
```

Comment:
close on Bolt won't happen until those transactions are completed. And if they have other writes coming in they won't be able to get to the shard because of the lock on store.